src/modules/flow/http-server/http-server.c

```diff
@@ -994,6 +1032,164 @@ direction_vector_open(struct sol_flow_node *node, void *data,
+        sol_blob_unref(mdata->value.blob);
+
+    mdata->value.blob = sol_blob_ref(blob);
+    SOL_NULL_CHECK(mdata->value.blob, -ENOMEM);
+    return 0;
+}
+
+static int
+blob_process_cb(struct http_data *mdata,
+    const struct sol_flow_packet *packet)
+{
+    int r;
+    struct sol_blob *blob;
+    const struct sol_flow_packet_type *packet_type;
+
+    packet_type = sol_flow_packet_get_type(packet);
+
+    if (packet_type == SOL_FLOW_PACKET_TYPE_JSON_OBJECT)
+        r = sol_flow_packet_get_json_object(packet, &blob);
+    else if (packet_type == SOL_FLOW_PACKET_TYPE_JSON_ARRAY)
+        r = sol_flow_packet_get_json_array(packet, &blob);
+    else
+        r = sol_flow_packet_get_blob(packet, &blob);
+
+    SOL_INT_CHECK(r, < 0, r);
+    return replace_json_blob(mdata, blob);
+}
+
+static void
+blob_send_packet_cb(struct http_data *mdata,
+    struct sol_flow_node *node)
+{
+    sol_flow_send_blob_packet(node,
+        SOL_FLOW_NODE_TYPE_HTTP_SERVER_BLOB__OUT__OUT, mdata->value.blob);
+}
+
+static int
+blob_handle_response_cb(struct sol_flow_node *node,
+    struct sol_http_request *request, struct sol_http_response *response,
+    bool *updated)
+{
+    int r;
+    struct http_data *mdata = sol_flow_node_get_private_data(node);
+
+    if (sol_http_request_get_method(request) == SOL_HTTP_METHOD_POST) {
+        struct sol_blob *blob = NULL;
+        void *blob_mem = NULL;
+        struct sol_http_param_value *param;
+        uint16_t i;
+
+        SOL_HTTP_PARAMS_FOREACH_IDX (sol_http_request_get_params(request),
+            param, i) {
+            if (param->type != SOL_HTTP_PARAM_POST_DATA)
+                continue;
+
+            blob_mem = malloc(param->value.data.value.len);
+            SOL_NULL_CHECK(blob_mem, -ENOMEM);
+            memcpy(blob_mem, param->value.data.value.data,
+                param->value.data.value.len);
+            blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL,
+                blob_mem, param->value.data.value.len);
+            break;
+        }
+
+        if (!blob && blob_mem) {
+            free(blob_mem);
+            SOL_WRN("Could not create a blob to hold the json data");
+            return -ENOMEM;
+        } else if (!blob && !blob_mem) {
+            SOL_WRN("The POST request has no POST data!");
+            return -ENOENT;
+        }
+
+        r = replace_json_blob(mdata, blob);
+        sol_blob_unref(blob);
+        SOL_INT_CHECK(r, < 0, r);
+        *updated = true;
+    }
+
+    if (!mdata->value.blob) {
+        response->response_code = SOL_HTTP_STATUS_NOT_FOUND;
+        return 0;
+    }
+
+    r = sol_buffer_append_bytes(&response->content,
+        (uint8_t *)mdata->value.blob->mem, mdata->value.blob->size);
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (!sol_http_param_add(&response->param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, HTTP_HEADER_CONTENT_TYPE_JSON))) {
+        SOL_WRN("Could not set the Content-Type:application/json");
+        return -ENOMEM;
```

Comment:
... the error treatment is done by who call this function. Forget these comments